src/hotspot/share/gc/shenandoah/shenandoahRootProcessor.inline.hpp

```diff
@@ -176,7 +176,7 @@ template <typename IsAlive, typename KeepAlive>
 void ShenandoahRootUpdater::roots_do(uint worker_id, IsAlive* is_alive, KeepAlive* keep_alive) {
   CodeBlobToOopClosure update_blobs(keep_alive, CodeBlobToOopClosure::FixRelocations);
   ShenandoahCodeBlobAndDisarmClosure blobs_and_disarm_Cl(keep_alive);
-  CodeBlobToOopClosure* codes_cl = ClassUnloading ?
+  CodeBlobToOopClosure* codes_cl = ClassUnloading && ShenandoahNMethodBarrier ?
```

Comment:
The hunk above has this in parentheses. Pick a style and stick to it, whether parentheses or not.